### PR TITLE
[rollout] feat: add optional tags parameter to release()

### PR DESCRIPTION
**Pull request by cavities12**
### What does this PR do?

Adds an optional `tags` parameter to `release()` across all rollout backends (SGLang, vLLM, TRT-LLM), matching the existing `resume(tags)` API introduced in #1911.

`release()` currently always releases both kv_cache and weights. With this change, callers can selectively release `["weights"]`, `["kv_cache"]`, or both `["kv_cache", "weights"]` (the default, preserving backward compatibility).

This completes the symmetry between `resume(tags)` and `release(tags)` that was left unfinished in #1911, enabling use cases like resyncing only weights after `update_weights()` without touching kv_cache.

### Checklist Before Starting

- [x] Search for similar PRs: https://github.com/verl-project/verl/pulls?q=is%3Apr+release+tags — #1911 added tag-based `resume()` but left `release()` hardcoded
- [x] Format the PR title as `[{modules}] {type}: {description}`

### Test

15 unit tests covering all 3 backends:
- Default (None) releases both — backward compatible
- Selective release (`["weights"]`, `["kv_cache"]`)
- Unknown tag validation (`ValueError`)
- Edge cases (non-leader noop, free_cache_engine disabled)
- vLLM: `["weights"]`-only raises `NotImplementedError` (sleep levels don't support it)

```
tests/workers/rollout/test_release_tags.py .... 15 passed in 0.03s
```

### API and Usage Example

```python
# Before (still works — backward compatible)
await rollout.release()  # releases both kv_cache and weights

# After — selective release
await rollout.release(tags=["weights"])    # release only weights
await rollout.release(tags=["kv_cache"])   # release only kv_cache
```

### Design & Code Changes

| File | Change |
|------|--------|
| `base.py` | ABC signature: `release(self, tags: list[str] \| None = None)` |
| `sglang_rollout.py` | Delegates tags to engine, validates unknown tags |
| `vllm_rollout.py` | Maps tags to sleep levels (1=kv_cache, 2=both), `NotImplementedError` for weights-only |
| `trtllm_rollout.py` | Resolves abstract `"weights"` to internal `_WEIGHTS_TAGS`, validates via set |
| `test_release_tags.py` | 15 async tests across all 3 backends |

All backends validate unknown tags with `ValueError`. Tags are normalized to `set()` to prevent duplicates.

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs). N/A — no existing rollout API reference doc.
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code.
- [x] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1).
- [X]  If your PR is related to the recipe submodule, update the reference. — N/A.

### CODE REVIEW

File: tests/workers/rollout/test_release_tags.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
+# Copyright 2025 Bytedance Ltd. and/or its affiliates
```
Comment (ETOgaosion) [inline tests/workers/rollout/test_release_tags.py:1]: Thanks for your contribution!
I suggest split vllm, trtllm and sglang tests as diffferent function with arguments, and seperate them to vllm/sgl/trtllm CI test using args to control.
So that when rollout engines upgrade, we can know the API changes at time.
